platform/security-filter-api/src/main/java/org/codice/ddf/platform/filter/HttpFilter.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.filter;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * An HttpFilter is an object that performs filtering tasks on either the request to a resource (a
+ * servlet or static content), or on the response from a resource, or both.
+ *
+ * <p>It is modelled after {@link javax.servlet.Filter} but its {@code doFilter} method accepts
+ * {@link HttpServletRequest} and {@link HttpServletResponse} arguments instead.
+ */
+public interface HttpFilter {
+
+  void doFilter(
```

Comment:
The Jetty Handlers take HttpServletRequests. In order to inject stuff into that chain, I needed to have a filter interface that takes HttpServletRequests as well. Oddly enough this is not already a thing.

Anything filters we used to inject via the delegate filter will need to be changed to implement this new interface. That will require changing the method signature and removing the init/destroy methods since those aren't used anymore.